File: Infrastructure/CellToScreenConverter.fs
namespace BattleCity.Infrastructure

open Avalonia.Data.Converters
open BattleCity.Model
open System

type CellToScreenConverter internal () =
    interface IValueConverter with
        member _.Convert (value, targetType, parameter, culture) : obj =
            (Convert.ToDouble(value) * GameField.CellSize) :> obj
        member _.ConvertBack (value, targetType, parameter, culture) : obj =
            raise <| NotSupportedException()
        
    static member Instance = CellToScreenConverter()

File: Infrastructure/CellToScreenConverter.cs
namespace BattleCity.Infrastructure;

using Avalonia.Data.Converters;
using Model;
using System;
using System.Globalization;

public class CellToScreenConverter : IValueConverter {
    public static CellToScreenConverter Instance { get; } = new();

    public object Convert(object value, Type targetType, object parameter, CultureInfo culture) =>
        System.Convert.ToDouble(value) * GameField.CellSize;

    public object ConvertBack(object value, Type targetType, object parameter, CultureInfo culture) =>
        throw new NotSupportedException();
}